test/add_test.go

```diff
@@ -294,7 +294,7 @@ func getWorkflows(dir string) map[v1alpha1.Type]map[v1alpha1.Action][]v1alpha1.P
 	var promise v1alpha1.Promise
 	ExpectWithOffset(1, yaml.Unmarshal(promiseYAML, &promise)).To(Succeed())
 
-	pipelines, err := v1alpha1.NewPipelinesMap(&promise, ctrl.LoggerFrom(context.Background()))
+	pipelines, err := v1alpha1.NewPipelinesMap(&promise, logr.FromContextOrDiscard(context.Background()))
```

Comment:
Is this a new pattern we want to use? Not blocking, but I can't see why we made this change in this PR and would like to keep consistent where possible especially since we have a refactoring coming and consistency will make that easier.